components/fxa-client/src/telemetry.rs

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// A somewhat mixed-bag of all telemetry we want to collect. The idea is that
+// the app will "pull" telemetry via a new API whenever it thinks there might
+// be something to record.
+// It's considered a temporary solution until either we can record it directly
+// (eg, via glean) or we come up with something better.
+
+use serde_derive::*;
+
+// First, constituent parts
+#[derive(Debug, Serialize)]
+pub struct SentReceivedTab {
```

Comment:
I wonder if having separate `SentTab` and `ReceivedTab` structs would help avoid the kind of awkward naming thing here (or perhaps even `SendCommand` and `ReceivedCommand` to line up more directly with the eventual event names).